### PR TITLE
Fix broken directory creation for firmware deltas

### PR DESCRIPTION
**Pull request by lawik**
This also improves the error information and logging around deltas.

It does not hook them up to be enabled. This code is currently dead.

### CODE REVIEW

**Comment by joshk** [comment]
Lars and I went on a deep dive into Delta land and can confirm that no code path generates or uses deltas.

So right now we have a tick box to enable delta firmware updates, but it does nothing.

As a note for later, we should look into changing all the `{_, 0}` matches as these raise match errors, which makes logging difficult and flow control a harder.

**Comment by oestrich** [comment]
Yeah I deleted the part that sends delta updates to devices a long time ago, smartrent needs some kind of work on firmwares to enable them (something with encryption? i dunno) so I haven't been pressed to add it back. I thought they were still generated, but I might have deleted that as well haha

**Comment by fhunleth** [comment]
It looks like there was a big misunderstanding. We'll need to get this feature back.

File: lib/nerves_hub/firmwares.ex [hunk 1]
```diff
@@ -313,10 +317,19 @@ defmodule NervesHub.Firmwares do
...
+            "Created firmware delta between #{source_firmware.uuid} and #{target_firmware.uuid}, successfully."
+          )
+
           firmware_delta
         else
           {:error, error} ->
             delta_updater().cleanup_firmware_delta_files(firmware_delta_path)
+
+            Logger.error(
+              "Failed to create firmwa delta between #{source_firmware.uuid} and #{target_firmware.uuid}: #{inspect(error)}"
```
Comment (fhunleth) [inline lib/nerves_hub/firmwares.ex:330]:
```suggestion
              "Failed to create firmware delta between #{source_firmware.uuid} and #{target_firmware.uuid}: #{inspect(error)}"
```

**Review by fhunleth — APPROVED** [review]

**Comment by oestrich** [comment]
For reference, this is the commit and code block that didn't survive a rewrite of the `resolve_update` function as I was figuring out what was going on in NervesHub at the time and trying to make deployments more coherent. 

https://github.com/nerves-hub/nerves_hub_web/commit/1c39a4f5ed154cf81dd22f6ffcf1cda4281b12c0#diff-03454c312d5c891b1ae7d916a1eb171cdf023ca00f9aa5155ea5633f7c5cbe81L525-L591

**Pull request merged**